### ISSUE TITLE
Setting RegressionTables methods and fields instead of StatsAPI ones

### ISSUE THREAD

**Issue by jsimons8** [jmboehm/RegressionTables.jl#153]
For adapting my own package to play nice with `RegressionTables`, I notice that a formula is mandatory in the new version. However, a formula per se makes little sense because the data would not have columns of hidden factors

A workaround was to construct a formula and pass it along:

```
f = @formula(Incidence ~ Level)
```
That formula can be passed (in place of formula_schema). However, I already have a vector of parameter names which are not really column names but a different type of parameter not related to any data set but to the model itself. I suppose, I could convert a vector of strings to a formula. Is there a workaround somehow that is simple to use that avoids the formula object altogether?

**Comment by junder873** [comment]
No, it is not necessary to have a formula object. It just needs `_coefnames` to be defined for your special type:
```julia
RegressionTables._coefnames(rr::MyModel) = RegressionTables.get_coefname($vector_of_strings_from_model)
# or
RegressionTables._coefnames(rr::MyModel) = string.($vector_of_strings_from_model)
```

Having `@formula` defined is useful for picking up interaction terms, but if that is not relevant in your situation, you do not need it.

**Comment by jsimons8** [comment]
Thank you for your response. That indeed does the trick.

I noticed the lines that did this previously have become obsolete?

```
StatsAPI.coefnames(m::MyModel) = m.coefnames;
StatsAPI.responsename(m::Attempt.FixedVarianceLocalLevel) = m.responsename
```

Does that mean more generally that the `StatsAPI` technology is obsolete and I should define `RegressionTables` methods instead? If that is true, is there a page somewhere that I can find all the methods so I can replace my definitions of `StatsAPI` methods.

**Comment by junder873** [comment]
The goal of RegressionTables is that if you fully implement StatsAPI on your custom type, then RegressionTables should "just work". You point out at least one bug and an oversight in that, so thank you. There also might be a few exceptions to that, such as not relying directly on `StatsAPI.coefnames`, but there should be a good reason (allowing labels on individual parts of interaction terms and changing how interactions are displayed). Even for these exceptions, RegressionTables should ideally still rely on the StatsAPI in some way (those functions rely on `StatsAPI.formula`).

More generally, comparing the StatsAPI is the standard throughout Julia. For example, if you had some other package that calculated some result on your custom regression model, it would probably rely on StatsAPI to do so. Then you would only need to implement one API and both statistics package and RegressionTables would work.

The bug and the oversight:
- Defining `StatsAPI.bic` should work; the RegressionTables `BIC` function is using `aaic` for some reason there. I will fix this.
- The default should be to use `StatsAPI.confint`, not create a custom implementation. I will need to make sure there was not some reason this was implemented differently. For now, you need to define `ConfInt(rr::MyStatsModel, k::Int; level=0.95, standardize=false, vargs...)` (line 507 of [this file](https://github.com/jmboehm/RegressionTables.jl/blob/master/src/RegressionStatistics.jl)) to make confidence interval work.

Finally, there are a few places where the documentation refers to the StatsModels API. This is sort of fuzzy because the StatsModels API generally implements StatsAPI (with some exceptions, `formula` is a StatsModels item). I am happy to take suggestions on making the documentation clearer/more precise.

**Comment by jsimons8** [comment]
Great, thank you so much! Once I get this done, I will write a little help file for people to use this based on my experiences and it would be good if that could be integrated into the documentation. It's a great package and it would be nice to see more statisticians use it.

Re: confidence intervals, I got them to work with this line:
```
function RegressionTables.ConfInt(m::Attempt.FixedVarianceLocalLevel, k::Int; level=0.95, standardize=false, vargs...)
    RegressionTables.ConfInt((1,3))
end
```
where the tuple for now is a placeholder of course.

**Comment by junder873** [comment]
I definitely appreciate any help with the documentation!

**Comment by junder873** [comment]
Both of those issues you pointed out should be fixed on the latest version.

**Comment by jsimons8** [comment]
Ok, so I can use the confidence interval API 
```
function StatsAPI.confint(m::MyModel; level::Real = 0.95)
    hcat([1],[3])
end
```

```
StatsAPI.bic
```
for the Bayesian information criterion?

**Comment by junder873** [comment]
On v0.7.5, it should (please let me know if it doesn't).

Just as a clarification though, the `StatsAPI.confint` is expecting a matrix with `nrow = length(coef)`, so
```julia
function StatsAPI.confint(m::MyModel; level::Real=0.95)
    hcat(fill(1, length(coef(m)), fill(3, length(coef(m)))
end
```
Will not cause errors.

**Comment by jsimons8** [comment]
I can confirm that using the `StatsAPI.confint` works now.
